coast/TIDEGAUGE.py

```diff
@@ -365,70 +369,58 @@ def plot_on_map_multiple(cls,tidegauge_list, color_var_str = None):
-        x = np.array(self.dataset.time)
-        y = np.array(self.dataset[var_name])
-        if qc_colors:
-           qc = np.array(self.dataset.qc_flags)
-        # Use only values between stated dates
-        start_index = 0
-        end_index = len(x)
-        if date_start is not None:
-            date_start = np.datetime64(date_start)
-            start_index = np.argmax(x>=date_start)
-        if date_end is not None:
-            date_end = np.datetime64(date_end)
-            end_index = np.argmax(x>date_end)
-        x = x[start_index:end_index]
-        y = y[start_index:end_index]
-        if qc_colors:
-           qc = qc[start_index:end_index]
-        
-        # Plot lines first if needed
-        if plot_line:
-            plt.plot(x,y, c=[0.5,0.5,0.5], linestyle='--', linewidth=0.5)
-        
-        # Two plotting routines for whether or not to use qc flags.
-        if qc_colors:
-            size = 5
-            fig = plt.figure(figsize=(10,10))
-            ax = plt.scatter(x[qc==0], y[qc==0], s=size)
-            plt.scatter(x[qc==1], y[qc==1], s=size)
-            plt.scatter(x[qc==2], y[qc==2], s=size)
-            plt.scatter(x[qc==3], y[qc==3], s=size)
-            plt.scatter(x[qc==4], y[qc==4], s=size)
-            plt.grid()
-            plt.legend(['No QC','Correct','Interpolated','Doubtful','Spike'],
-                       loc='upper left', ncol=5)
-            plt.xticks(rotation=45)
-        else:
-            fig = plt.figure(figsize=(10,10))
-            ax = plt.scatter(x,y)
-            plt.grid()
-            plt.xticks(rotation=65)
+        fig = plt.figure(figsize=(10,10))
+        # Check input is a list (even for one variable)
+        if type(var_list) is str:
+            var_list = [var_list]
+        
+        for var_str in var_list:
+            dim_str = self.dataset[var_str].dims[0]
+            x = np.array(self.dataset[dim_str])
+            y = np.array(self.dataset[var_str])
+    
+            # Use only values between stated dates
+            start_index = 0
+            end_index = len(x)
+            if date_start is not None:
+                date_start = np.datetime64(date_start)
+                start_index = np.argmax(x>=date_start)
+            if date_end is not None:
+                date_end = np.datetime64(date_end)
+                end_index = np.argmax(x>date_end)
+            x = x[start_index:end_index]
+            y = y[start_index:end_index]
+            
+            # Plot lines first if needed
+            if plot_line:
+                plt.plot(x,y, c=[0.5,0.5,0.5], linestyle='--', linewidth=0.5)
+            
+            ax = plt.scatter(x,y, s=10)
             
+        plt.grid()
+        plt.xticks(rotation=45)
+        plt.legend(var_list)
         # Title and axes
         plt.xlabel('Date')
-        plt.ylabel(var_name + ' (m)')
-        plt.title(var_name + ' at site: ' + self.dataset.site_name)
+        plt.ylabel(var_str + ' (m)')
```

Comment:
It looks like the ylabel will be the last variable in var_list. Is this intended?

Reply:
Not any more. Will remove.